tokenizer.go

```diff
@@ -67,6 +88,155 @@ func FromFile(path string) (*Tokenizer, error) {
 	return &Tokenizer{tokenizer: tokenizer}, nil
 }
 
+// LoadTokenizerFromHuggingFace downloads necessary files and initializes the tokenizer.
+// Parameters:
+//   - modelID: The Hugging Face model identifier (e.g., "bert-base-uncased").
+//   - destination: Optional. If provided and not nil, files will be downloaded to this folder.
+//     If nil, a temporary directory will be used.
+//   - authToken: Optional. If provided and not nil, it will be used to authenticate requests.
+func LoadTokenizerFromHuggingFace(modelID string, destination, authToken *string) (*Tokenizer, error) {
+	if strings.TrimSpace(modelID) == "" {
+		return nil, fmt.Errorf("modelID cannot be empty")
+	}
+
+	// Construct the model URL
+	modelURL := fmt.Sprintf("%s/%s/resolve/main", baseURL, modelID)
+
+	// Determine the download directory
+	var downloadDir string
+	if destination != nil && *destination != "" {
+		downloadDir = *destination
+		// Create the destination directory if it doesn't exist
+		err := os.MkdirAll(downloadDir, os.ModePerm)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create destination directory %s: %w", downloadDir, err)
+		}
+	} else {
+		// Create a temporary directory
+		tmpDir, err := os.MkdirTemp("", "huggingface-tokenizer-*")
+		if err != nil {
+			return nil, fmt.Errorf("error creating temporary directory: %w", err)
+		}
+		downloadDir = tmpDir
+	}
+
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(tokenizerFiles))
```

Comment:
let's use unbuffered channel: https://github.com/uber-go/guide/blob/master/style.md#channel-size-is-one-or-none

using buffered channels tends to hide concurrency issues if present